4.gene_expression_signatures/scripts/0.latent_dimension_dataframe.py

```diff
@@ -0,0 +1,283 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pathlib 
+import optuna
+import pandas as pd
+import numpy as np
+import random
+import torch
+import sys
+import blitzgsea as blitz
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+from sklearn.decomposition import PCA, FastICA, NMF
+from sklearn.preprocessing import MinMaxScaler
+from torch.utils.data import DataLoader, TensorDataset
+from scipy.stats import ttest_ind
+
+script_directory = pathlib.Path("../2.train-VAE/utils/").resolve()
+sys.path.insert(0, str(script_directory))
+from betavae import BetaVAE, train_vae, weights
+from betatcvae import BetaTCVAE, train_tc_vae, tc_weights
+from vanillavae import VanillaVAE, train_vvae, vanilla_weights
+from optimize_utils import get_optimize_args, objective, get_optimizer
+from optimize_utils_tcvae import get_optimize_args_tc, objective_tc, get_optimizer_tc
+from optimize_utils_vvae import get_optimize_args_vvae, objective_vvae, get_optimizer_vvae
+
+script_directory = pathlib.Path("../utils/").resolve()
+sys.path.insert(0, str(script_directory))
+from data_loader import load_train_test_data, load_model_data
+
+
+# In[2]:
+
+
+# Load command line arguments
+args = get_optimize_args()
+tc_args = get_optimize_args_tc()
+vvae_args = get_optimize_args_vvae()
+
+# Load data
+data_directory = pathlib.Path("../0.data-download/data").resolve()
+
+train_df, test_df, val_df, load_gene_stats = load_train_test_data(
+    data_directory, train_or_test="all", load_gene_stats=True, zero_one_normalize=True
+)
+train_data = pd.DataFrame(train_df)
+
+dependency_file = pathlib.Path(f"{data_directory}/CRISPRGeneEffect.parquet").resolve()
+gene_dict_file = pathlib.Path(f"{data_directory}/CRISPR_gene_dictionary.parquet").resolve()
+dependency_df, gene_dict_df= load_model_data(dependency_file, gene_dict_file)
+gene_dict_df = pd.DataFrame(gene_dict_df)
+
+
+# In[3]:
+
+
+#Load weight data for VAEs
+data_directory = pathlib.Path("../0.data-download/data").resolve()
+weight_df = load_train_test_data(
+    data_directory, train_or_test="train"
+)
+
+gene_list_passed_qc = gene_dict_df.loc[
+    gene_dict_df["qc_pass"], "dependency_column"
+].tolist()
+
+weight_data = weight_df.filter(gene_list_passed_qc, axis=1)
+weight_data.head()
+
+
+# In[4]:
+
+
+# Convert dataframes to tensors
+train_tensor = torch.tensor(train_df, dtype=torch.float32)
+test_tensor = torch.tensor(test_df, dtype=torch.float32)
+val_tensor = torch.tensor(val_df, dtype=torch.float32)
+
+
+# In[5]:
+
+
+output_dir = pathlib.Path("./results").resolve()
+output_dir.mkdir(parents=True, exist_ok=True)
+
+
+# In[6]:
+
+
+# Function to perform dimensionality reduction and extract z matrices for sklearn models
+def extract_z_matrix_sklearn(model, data, model_name, num_components, output_dir):
```

Comment:
neat function! However, it should be defined at the beginning on the notebook (after imports)

Reply:
Fixed!